### PR TITLE
Katello 3.1 backports and adding modules

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: Puppetfile [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-mod 'katello/service_wait',     :git => 'https://github.com/Katello/puppet-service_wait'
+mod 'katello/candlepin',        '1.0.1'
+mod 'katello/capsule',          '3.0.0' 
+mod 'katello/certs',            '0.6.0'
+mod 'katello/common',           '0.1.3'
+mod 'katello/crane',            '1.0.0'
+mod 'katello/katello',          '5.0.1'
+mod 'katello/pulp',             '3.3.1'
+mod 'katello/qpid',             '1.2.2' 
+mod 'katello/service_wait',     '0.1.2'
```
Comment (stbenjam) [inline Puppetfile]: Hm, when did we start pinning these so strictly? I see it in 3.0 too, but the way foreman does it, and we did it for previous releases is to pin to a minor.

e.g., https://github.com/Katello/katello-installer/blob/KATELLO-2.4/Puppetfile#L29-L41

It makes it easier for smaller bug fixes I think to just re-run librarian and pull in the latest patch release.

**Comment by ehelms** [comment]
You'll need to rebuild the modules directory using `rake setup_local` so that you are omitting modules that are provides by the foreman-installer.

**Comment by jlsherrill** [comment]
@ehelms did you have any response to @stbenjam's question?  I just followed your 3.0 work

**Comment by ehelms** [comment]
Can if you want - since these are our modules and we do deliberate big
fixing I have leaned towards locking them down and pulling specific updates
when the release warrants it vs keeping it open for updates every time the
installer is rebuilt.
On Jun 23, 2016 10:46 AM, "Justin Sherrill" <notifications@github.com>
wrote:

@ehelms <https://github.com/ehelms> did you have any response to @stbenjam
<https://github.com/stbenjam>'s question? I just followed your 3.0 work

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/Katello/katello-installer/pull/369#issuecomment-228073278>,
or mute the thread
<https://github.com/notifications/unsubscribe/AAR586MeS4YzHpGN23ZMlKCrT0Ma89kPks5qOpwugaJpZM4I81ND>
.

**Comment by stbenjam** [comment]
It's still not automatic, for releases you have run the rake task again.  With such strict locking, it's 2 steps, update the Puppetfile then the rake task.  The end result is the same so it doesn't much matter to me, was just curious.

**Comment by ehelms** [comment]
ACK

**Pull request merged**